packages/noya-state/src/reducers/symbolsReducer.ts

```diff
@@ -0,0 +1,175 @@
+import Sketch from '@sketch-hq/sketch-file-format-ts';
+import produce from 'immer';
+import { getSymbols } from '../selectors/themeSelectors';
+import { getCurrentTab } from '../selectors/workspaceSelectors';
+import { ApplicationState } from './applicationReducer';
+
+export type SymbolsAction =
+  | [type: 'setAdjustContentOnResize', value: boolean]
+  | [type: 'setHasBackgroundColor', value: boolean]
+  | [type: 'setBackgroundColor', value: Sketch.Color]
+  | [type: 'setIncludeBackgroundColorInExport', value: boolean]
+  | [type: 'setIncludeBackgroundColorInInstance', value: boolean]
+  | [type: 'setGroupLayout', value: Sketch.InferredLayoutAxis | undefined]
+  | [type: 'setLayoutAnchor', value: Sketch.InferredLayoutAnchor]
+  | [type: 'setLayoutAnchor', value: Sketch.InferredLayoutAnchor]
+  | [type: 'setMinWidth', value: number]
+  | [type: 'setAllowsOverride', value: boolean];
+export function symbolsReducer(
+  state: ApplicationState,
+  action: SymbolsAction,
+): ApplicationState {
+  switch (action[0]) {
+    case 'setAdjustContentOnResize': {
+      const [, value] = action;
+      const filter =
+        getCurrentTab(state) === 'canvas'
+          ? state.selectedObjects
+          : state.selectedSymbolsIds;
+
+      return produce(state, (draft) => {
+        const symbol = getSymbols(draft).filter((symbol) =>
+          filter.includes(symbol.do_objectID),
+        )[0];
```

Comment:
Checking for `[0]` should be safe given that we should never call this without a selection, but since eventually we want to handle multiple symbols, and since it's not really more effort, I might just update all symbols in a forEach.

```ts
const symbols = ...

symbols.forEach(symbol => {
  symbol.resizesContent = value;
})
```